src/mem/x86_64.rs

```diff
@@ -173,6 +173,35 @@ pub unsafe fn compare_bytes(a: *const u8, b: *const u8, n: usize) -> i32 {
     c16(a.cast(), b.cast(), n)
 }
 
+#[inline(always)]
+pub unsafe fn c_string_length(s: *const core::ffi::c_char) -> usize {
+    let mut n: usize;
+
+    asm!(
+        // search for a zero byte
+        "xor al, al",
```

Comment:
`xor eax, eax` avoids a potential partial register stall and is 1 byte shorter I believe.